### DISCUSSION TITLE
How to not affect the second form inside a component even though I used a custom register

### DISCUSSION THREAD

**Discussion by nani1008** [react-hook-form/react-hook-form#5237] [Q&A] [answered]
So I have a 2 forms which is affecting the first form if I submit the second form, The second form is inside a component. Here is the structure

```
<App (first form here) I used useFormContext here>
   <Component  (second form here)/>
</App> 
```

I dont want the first form(on the parent) to be triggered if i used onSubmit on the second form. I already used a custom register and handleSubmit on the component which is: 

```
  const methods = useFormContext();
  const {
    register: customRegister,
    handleSubmit: customHandleSubmit,
  } = methods;
```

But it is still affecting the first form if I submit the second form, any solution to this?

**Comment by nani1008** [comment]
Please help im stuck any suggestions will do @bluebill1049

**Reply by bluebill1049** [reply]
perhaps you should have two useForm instance?

**Reply by nani1008** [reply]
@bluebill1049 so on the first useForm which is the main component <App> i'll set a custom register also?

**Comment by thanh-nguyen-95** [comment, marked as answer]
If you use `<button type="submit">` to submit the second form, try adding `event.stopPropagation()` in the `<form onSubmit={...}>` to prevent the `Event` from bubbling to the parent form.

I personally use 2 `useForm` instances as @bluebill1049 said: 1 passed through `useFormContext` and 1 used directly inside child component.

**Reply by nani1008** [reply]
I cant trigger the onSubmit if I add `event.stopPropagation()`

**Reply by thanh-nguyen-95** [reply]
```js
const methods = useFormContext();

const {
  register: customRegister,
  handleSubmit: customHandleSubmit,
} = methods;
```

Doing this will only renames the `register` and `handleSubmit` (giving them an alias), not creating a new instance of them. In your case, you should use 2 form instances like @bluebill1049 said.

An example:

```jsx
function ParentForm() {
  const methods = useForm()
  
  return (
    <FormProvider {...methods}>
      <form onSubmit={methods.handleSubmit(console.log)}>
        <ChildForm />
        <button type="submit">Submit parent form</button>
      </form>
    </FormProvider>
  )
}

function ChildForm() {
  const parentFormMethods = useFormContext()
  const childFormMethods = useForm()

  return (
    <form onSubmit={ev => {
      ev.stopPropagation()  // this stops the event from bubbling to parent form
      childFormMethods.handleSubmit(console.log)
    }}>
      <button type="submit">Submit child form</button>
    </form>
  )
}
```
